### PR TITLE
Sw test

### PR DESCRIPTION
**Pull request by zFernand0**

### CODE REVIEW

**Comment by zFernand0** [comment]
ℹ️ git tag and git push need to be uncommented
ℹ️ --dry-run needs removed from npm publish

**Comment by zFernand0** [comment]
Here is a workgin build.
The archived npm-shrinkwrap is what was produced after groovy.prettifying the object.
Maybe a future alternative is to `JsonSlurper` it, but a better one is to move to Octorelease! 😋

File: src/org/zowe/pipelines/nodejs/NodeJSPipeline.groovy [hunk 1]
```diff
@@ -955,10 +965,10 @@ class NodeJSPipeline extends GenericPipeline {
                     if (!changeInfo.isPullRequest) {
                         // Add package and package lock to the commit tree. This will not fail if
                         // unable to add an item for any reasons.
-                        steps.sh "git add package.json package-lock.json --ignore-errors || exit 0"
+                        steps.sh "git add package.json package-lock.json npm-shrinkwrap.json --ignore-errors || exit 0"
                         if (isLernaMonorepo) {
                             runForEachMonorepoPackage(LernaFilter.ALL) {
-                                steps.sh "git add package.json --ignore-errors || exit 0"
+                                steps.sh "git add package.json npm-shrinkwrap.json --ignore-errors || exit 0"
```
Comment (t1m0thyj) [inline src/org/zowe/pipelines/nodejs/NodeJSPipeline.groovy:971]: Do we want to stage `npm-shrinkwrap.json` files that are inside a subpackage directory of a monorepo? I believe we should omit `npm-shrinkwrap.json` files from this command, for the same reason that we omitted `package-lock.json`.

File: src/org/zowe/pipelines/nodejs/NodeJSPipeline.groovy [hunk 2]
```diff
@@ -1080,6 +1090,50 @@ class NodeJSPipeline extends GenericPipeline {
...
 
+    /**
+     * Remove all dev dependencies from the shrinkwrap file before publishing.
+     *
+     * @param body The closure to execute in when temporarily rewriting the shrinkwrap
+     */
+    protected void rewriteShrinkwrap(Closure body) {
+        def swJson = "MISSING"
+        try {
+            steps.sh "pwd;ls -al"
```
Comment (t1m0thyj) [inline src/org/zowe/pipelines/nodejs/NodeJSPipeline.groovy:1101]: Is this step for debugging purposes only, and if so should it be removed?

File: src/org/zowe/pipelines/nodejs/NodeJSPipeline.groovy (hunk 1 above)
Comment (zFernand0) [reply src/org/zowe/pipelines/nodejs/NodeJSPipeline.groovy:971]: :facepalm: you are right 😋

**Review by t1m0thyj — APPROVED** [review]
Worth noting that this requires the Jenkins agent to have `jq`, if the component being deployed has a "npm-shrinkwrap.json" file. I think this is ok, since the Zowe Jenkins agent and our internal agents all have it.

Thanks for implementing the shrinkwrap rewrite and going through the tedious parts of testing it 🙂

**Review by awharn — APPROVED** [review]

**Pull request merged**